### DISCUSSION TITLE
semantic tag

### DISCUSSION THREAD

**Discussion by michoball** [wanted-9th-3team/frontend-study#72] [프론트 기술 질문 공간]
semantic tag 를 사용해야되는 이유와 알고있는 semantic tag 들을 나열하고 그 쓰임에 대해 서술하시오

**Comment by michoball** [comment]
시멘틱 태그의 사용은 해당 앱의 seo를 좋게 할 뿐만 아니라 각 브라우저나 기기가 페이지를 이해하기 좋게 합니다. 
 작성한 HTML, ( 리액트에서는 ) JSX에서 각 태그의 목적은 분명히해 가독성을 좋게 합니다. 

header: 페이지의 맨 
nav:  네이게이션 링크가 모인 컨테이
section: 페이지의 한 부분
article: 페이지의 제일 주된 정보가 들어있는 부분
aside: 사이드 바와 같이 페이지 내 주된 정보와 별개인 부분
footer :페이지의 맨 밑 
main: 메인 페이지를 특정지을 때 특히 앱에서 유일무이하게 사용됨

이외에도 몇가지 시멘틱 태그들이 더 존재합니다. 
용도에 맞게 시멘틱 태그를 사용하는 것이 중요합니다.

**Comment by WooGie911** [comment]
semantic tag를 사용하면 

1. SEO(Search Engine Optimization) 최적화에 유리합니다. 
검색 엔진이 태그의 목적에 부합하게 설계되어있는 구조의 사이트에서 더욱 빨리 효율적으로 정보를 파악할 수 있어 검색 결과의 노출에 유리할 수 있게 해줍니다.

2. 웹 접근성에 효율적입니다.
일반적인 브라우저에서는 차이가 없지만 스크린리더(시각장애인을 위한 웹 서핑 프로그램)과 같은 환경에서는 웹 접근성과 사용성을 향상시켜 줍니다.

3. 유지보수가 용이해집니다.
무분별한 div사용으로 관리가 어려워지는 문제에서 벗어나 태그의 이름만 보고도 어떤 영역인지 바로 확인이 가능하며 해당 태그 영역의 특성에 맞는 작업을 구분하여 유지 보수에 용이해집니다.

여러 시멘틱 태그가 존재하지만 실제로 사용해본 태그는 
header (머리글, 제목, 헤더), 
nav (네이게이션, 목차, 리스트 등 다른 페이지로의 이동을 위한 링크 공간), 
footer (바닥글, 문서 하단에 들어가는 정보),
main (이름처럼 문서 <body>의 중심 주제, 주요 내용 또는 응용 프로그램의 중심 기능과 직접 관련되어나 확장되는 콘텐츠),
section (말그대로 주제, 카테고리 별로 섹션을 구분하는 용도의 태그로 주로 사용. 같은 테마를 가진 여러개의 콘텐츠의 그룹화)
정도 입니다.

**Comment by javascriptbiggosoo** [comment]
시맨틱 태그의 존재만 알고 있다면 사실 적용 자체는 모든 시맨틱 태그를 암기하지 않고 그때그때 적합한 것을 검색해 찾아쓰면 된다고들 했지만 결국 익숙해지지 않으면 시간이 걸리는 작업이었는데요.
요즘 제가 사용하는 방식은 이렇습니다.
1. 일단 개발함
2. 개발자도구에서 element를 전부 복사해서 챗gpt에게 시맨틱지 적절한지 물어봄

### 실제로 얻은 답변: (원본 코드에서는 body 이하 코드에서 nav를 제외하면 모두 div를 사용했었는데 header와 main, section 사용을 권해줌)
Sure, I'd be happy to help you enhance the semantic structure of your HTML. Here's a revised version of your HTML code with more semantic tags:

```html
<html lang="en">
<head>
    <!-- Head content -->
</head>
<body>
    <header>
        <nav class="sc-fPXMVe gqGCiq" style="background-color: rgba(0, 0, 0, 0);">
            <!-- Navigation content -->
        </nav>
    </header>
    <main>
        <div class="sc-dLMFU">
            <section class="sc-jlZhew dTiMuE">
                <!-- Section content -->
            </section>
            <section class="sc-aXZVg fcCJKF">
                <!-- Section content -->
            </section>
            <section class="sc-aXZVg fcCJKF">
                <!-- Section content -->
            </section>
        </div>
    </main>
    <script type="module" src="/src/main.tsx?t=1688028504529"></script>
</body>
</html>
```

In this version, I've added the `<header>`, `<main>`, and `<section>` tags to provide a clearer structure to the page:

- `<header>`: This tag is used to contain introductory content or navigational links. In your case, I've placed the navigation bar inside this tag.
- `<main>`: This tag is used for the dominant content of the body of a document. In your case, I've placed the main content of the page inside this tag.
- `<section>`: This tag represents a standalone section of a document, which typically contains a block of related content. I've used this tag to group related content together.

Please note that these are just suggestions, and the actual semantic structure may vary based on the specific content and design of your webpage.
